structs.asm

```diff
@@ -173,6 +173,42 @@ STRUCT_FIELD_TYPE equs STRSUB("\2", 2, 1)
 NB_FIELDS = NB_FIELDS + 1
 ENDM
 
+; dunion union_name
+; Defines an Alias for the next field
+dunion: MACRO
+    IF !DEF(STRUCT_NAME) || !DEF(NB_FIELDS)
+        FAIL "Please start defining a struct, using `define_struct`"
+    ENDC
+
+    get_nth_field_info NB_FIELDS
+    ; Set field name (keep in mind `STRUCT_FIELD_NAME` is *itself* an EQUS!)
+STRUCT_FIELD_NAME equs "\"\1\""
+    PURGE STRUCT_FIELD_NAME
+
+    ; Set field offset
+STRUCT_FIELD RB 0
+    ; Alias this in a human-comprehensive manner
+STRUCT_FIELD_NAME equs "{STRUCT_NAME}_\1"
+STRUCT_FIELD_NAME = STRUCT_FIELD
+
+    ; Compute field size
+CURRENT_RS RB 0
+STRUCT_FIELD_SIZE = CURRENT_RS - STRUCT_FIELD
+
+    ; Set properties
+STRUCT_FIELD_NBEL = 0
+STRUCT_FIELD_TYPE equs STRSUB("RB", 2, 1)
+
+    PURGE STRUCT_FIELD
+    PURGE STRUCT_FIELD_NAME
+    PURGE STRUCT_FIELD_TYPE
+    PURGE STRUCT_FIELD_NBEL
+    PURGE STRUCT_FIELD_SIZE
+    PURGE CURRENT_RS
+
+NB_FIELDS = NB_FIELDS + 1
```

Comment:
I don't think `dunion` should be incrementing the total field count, since that's how many arguments `dstruct` expects.

Something like this would work for v2 (haven't tested it):

```
; Defines an alias for the next field.
MACRO dunion ; field_name
    IF !DEF(STRUCT_NAME) || !DEF(NB_FIELDS)
        FAIL "Please start defining a struct, using `struct`"
    ENDC

    get_nth_field_info {STRUCT_NAME}, NB_FIELDS

    ; Set field name
    DEF {STRUCT_FIELD_NAME} EQUS "\1"
    ; Set field offset
    DEF {STRUCT_FIELD} RB 0
    ; Alias this in a human-comprehensible manner
    DEF {STRUCT_NAME}_\1 EQU {STRUCT_FIELD}
    ; Set properties
    DEF {STRUCT_FIELD_SIZE} EQU 0
    DEF {STRUCT_FIELD_NBEL} EQU 0
    DEF {STRUCT_FIELD_TYPE} EQUS "B"

    purge_nth_field_info
ENDM
```